windows/src/desktop/setup/UfrmDownloadProgress.pas

```diff
@@ -26,117 +25,70 @@
-    procedure HTTPStatus(Sender: THTTPUploader; const Message: string; Position, Total: Int64);  // I2855
+    procedure HTTPCheckCancel(Sender: TObject; var Cancel: Boolean);
+    procedure HTTPStatus(Sender: TObject; const Message: string; Position, Total: Int64);  // I2855
     property Callback: TDownloadProgressCallback read FCallback write FCallback;
     property Cancel: Boolean read FCancel;
   end;
 
 implementation
 
-{R *.dfm}
+{$R *.dfm}
 
 procedure TfrmDownloadProgress.cmdCancelClick(Sender: TObject);
 begin
   inherited;
   FCancel := True;
 end;
 
-procedure TfrmDownloadProgress.HTTPCheckCancel(Sender: THTTPUploader; var Cancel: Boolean);
+procedure TfrmDownloadProgress.FormCreate(Sender: TObject);
 begin
-  Cancel := FCancel;
-end;
-
-{procedure TfrmDownloadProgress.HTTPFileProgress(
-  Sender: THTTPUploader; const FileName: string; dwFileBytesSent,
-  dwLocalFileSize, dwSecondsToFileCompletion, dwOverallBytesSent,
-  dwOverallBytesTotal, dwSecondsToOverallCompletion, dwBytesPerSecond: DWord);
-begin
-  progress.Max := dwOverallBytesTotal;
-  progress.Position := dwOverallBytesSent;
-  progress.Update;
-  Application.ProcessMessages;
-end;}
-
-procedure TfrmDownloadProgress.HTTPStatus(Sender: THTTPUploader;
-  const Message: string; Position, Total: Int64);  // I2855
-var
-  msg: TMsg;
-begin
-  Max := Total;
-  if Max = 0 then
-  begin
-    Max := 100;
-    SendDlgItemMessage(Handle, IDC_PROGRESS1, PBM_SETRANGE, 0, MAKELONG(0, 100));
-  end;
-
-  SendDlgItemMessage(Handle, IDC_PROGRESS1, PBM_SETPOS, Position * 100 div Max, 0);
-  UpdateWindow(GetDlgItem(Handle, IDC_STATUS));
-  UpdateWindow(GetDlgItem(Handle, IDC_PROGRESS1));
-  Tnt_SetDlgItemText(IDC_STATUS, Message);
-
-  while PeekMessage(msg, 0, 0, 0, PM_REMOVE) do
-  begin
-    if not IsDialogMessage(Handle, msg) then
-    begin
-      TranslateMessage(msg);
-      DispatchMessage(msg);
-    end;
-  end;
+//  cmdCancel.Caption := MsgFromId(SKButtonCancel);
+//  Caption := MsgFromId(SKDownloadProgress_Title);
 end;
 
-function TfrmDownloadProgress.ProcessCommand(ID, NotificationCode: Integer;
-  hControl: HWND): Boolean;
-begin
-  Result := True;
-  case ID of
-    IDCANCEL: cmdCancelClick(Self);
-    else Result := False;
-  end;
-end;
-
-function TfrmDownloadProgress.DialogID: Integer;
+procedure TfrmDownloadProgress.FormShow(Sender: TObject);
 begin
-  Result := 101;
+  PostMessage(Handle, WM_USER, 0, 0); // Starts process after form displays
 end;
 
-procedure TfrmDownloadProgress.DlgMain(var Message: TMessage);
+procedure TfrmDownloadProgress.HTTPCheckCancel(Sender: TObject; var Cancel: Boolean);
 begin
-  case Message.Msg of
-    WM_USER_FormShown:
-      WMUserFormShown(Message);
-  else inherited;
-  end;
+  Cancel := FCancel;
 end;
 
-procedure TfrmDownloadProgress.FormShow(Sender: TObject);
+procedure TfrmDownloadProgress.HTTPStatus(Sender: TObject;
+  const Message: string; Position, Total: Int64);  // I2855
 begin
-  inherited;
```

Comment:
I see that this keyword was dropped when this method was repositioned within its source?

Reply:
Yes, the form is no longer an inherited form (as it was, a long time ago), so no need to call base methods (there aren't any!)